Orchestrator.py

```diff
@@ -191,25 +205,81 @@ def receive(self,frame):
         # simulation complete when there are no more frontier cells left
         if not self.cellsFrontier:
             self.simEngine.completeRun()
+
         log.debug(f'remaining frontiers are {self.cellsFrontier}')
 
         # update dotBotsView
-        dotBot['x']       = newX
-        dotBot['y']       = newY
-        log.debug(f'dotbot {dotBot} is at ( {newX},{newY} ) ')
+        dotBot['x']      = newX
+        dotBot['y']      = newY
+        dotBot['bumped'] = frame['bumped']
+
+        log.debug(f'dotbot {dotBot} is at ({newX},{newY})')
+
+        # if dotBot bumped
+        if frame['bumped']:
+            self.isDirectPathOpen = False
+
+        # assign target cell (destination)
+        if not dotBot['targetCell'] or dotBot['targetCell'] not in self.cellsFrontier:
```

Comment:
add parenthesis `if (not dotBot['targetCell']) or (dotBot['targetCell'] not in self.cellsFrontier):`